src/structured_data.clj

```diff
@@ -19,96 +27,199 @@
+      (contains-point? (rectangle [x1 y1] [x2 y2]) (point x4 y4)))
+  ))
+
+;(def china {:name "China Miéville", :birth-year 1972})
+;(def octavia {:name "Octavia E. Butler":birth-year 1947 :death-year 2006})
+;(def friedman {:name "Daniel Friedman" :birth-year 1944})
+;(def felleisen {:name "Matthias Felleisen"})
+
+;(def cities {:title "The City and the City" :authors [china]})
+;(def wild-seed {:title "Wild Seed", :authors [octavia]})
+;(def embassytown {:title "Embassytown", :authors [china]})
+;(def little-schemer {:title "The Little Schemer" :authors [friedman, felleisen]})
+;(def books [cities, wild-seed, embassytown, little-schemer])
 
 (defn title-length [book]
-  :-)
+  (count (book :title))
+  )
 
 (defn author-count [book]
-  :-)
+  (count (book :authors)))
 
 (defn multiple-authors? [book]
-  :-)
+  (not (= (count (book :authors)) 1))
+  )
 
 (defn add-author [book new-author]
-  :-)
+  (assoc book :authors (conj (:authors book) new-author))
+  )
 
 (defn alive? [author]
-  :-)
+  (not (contains? author :death-year))
+  )
 
 (defn element-lengths [collection]
-  :-)
+  (map (fn [x] (count x)) collection)
+  )
+
+;(defn second-elements [collection]
+;  (map (fn [x] (get x 1)) collection)
+;  )
+; exercise requires let so...
 
 (defn second-elements [collection]
-  :-)
+  (let [get-second (fn [x] (get x 1))] (map get-second collection)))
 
 (defn titles [books]
-  :-)
+  (map :title books)
+  )
+;; pregunta!
+;; (titles books) funciona
+;; (titles cities) no funciona
+;; (title [citites]) funciona
+;; veo que books es un vector y cities un mapa
+;; la pregunta seria, por que un mapa no puede ser parametro?
 
 (defn monotonic? [a-seq]
-  :-)
+  (or (apply >= a-seq) (apply <= a-seq))
+  )
 
 (defn stars [n]
-  :-)
+  (apply str (repeat n "*"))
+  )
 
 (defn toggle [a-set elem]
-  :-)
+  (if (contains? a-set elem)
+    (disj a-set elem) (conj a-set elem)
+  ))
 
 (defn contains-duplicates? [a-seq]
-  :-)
+  (not= (count a-seq) (count (set a-seq)))
+  )
 
 (defn old-book->new-book [book]
-  :-)
+  (assoc book :authors (set (:authors book)))
+  )
+
+;(def china {:name "China Miéville", :birth-year 1972})
+;(def octavia {:name "Octavia E. Butler"
+;              :birth-year 1947
+;              :death-year 2006})
+;(def friedman {:name "Daniel Friedman" :birth-year 1944})
+;(def felleisen {:name "Matthias Felleisen"})
+
+;(def cities {:title "The City and the City" :authors #{china}})
+;(def wild-seed {:title "Wild Seed", :authors #{octavia}})
+;(def embassytown {:title "Embassytown", :authors #{china}})
+;(def little-schemer {:title "The Little Schemer"
+;                     :authors #{friedman, felleisen}})
+
+;(def books [cities, wild-seed, embassytown, little-schemer])
 
 (defn has-author? [book author]
-  :-)
+  (contains? (:authors book) author)
+  )
 
 (defn authors [books]
-  :-)
+  (let [get-authors (fn [x] (:authors x))]
+    (apply clojure.set/union (map get-authors books))
```

Comment:
Dalee, grazie
El mar. 14, 2016 10:58 PM, "Gal Dolber" <notifications@github.com> escribió:

> In src/structured_data.clj
> <https://github.com/iloveponies/structured-data/pull/865#discussion_r56105776>
> :
>
> >
> >  (defn authors [books]
> > -  :-)
> > +  (let [get-authors (fn [x] (:authors x))]
> > +    (apply clojure.set/union (map get-authors books))
>
> Esta esta perfecto, un par de cositas.
>
> Acordate que aplicar una funcion a un solo argumento es lo mismo que usar
> de una esa funcion:
> (fn [x] (:authors x)) es igual a :authors
>
> Hay una funcion llamada mapcat que hace esto:
> (apply concat (map f colls))
> Osea, aplica un map y despues concatena todos los resultados. Aca la podes
> usar y obtener una respuesta asi:
>
> (def authors [books]
>     (mapcat :authors books))
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/iloveponies/structured-data/pull/865/files#r56105776
>